src/main/java/org/recordrobotics/charger/subsystems/Arm.java

```diff
@@ -0,0 +1,116 @@
+package org.recordrobotics.charger.subsystems;
+
+import org.recordrobotics.charger.RobotMap;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Arm extends SubsystemBase {
+	private WPI_TalonFX _originMotor = new WPI_TalonFX(RobotMap.Arm.ORIGIN_MOTOR_PORT);
+	private WPI_TalonFX _changeMotor = new WPI_TalonFX(RobotMap.Arm.CHANGE_MOTOR_PORT);
+
+	private static final FeedbackDevice SELECTED_SENSOR = FeedbackDevice.CTRE_MagEncoder_Relative;
+
+	private static final double FIRST_ARM_LENGTH = 30;
+	private static final double SECOND_ARM_LENGTH = 30;
+
+	// IN DEGREES
+	// negative is in the same direction as the rotation, positive is in the opposite
+	private static final double ORIGIN_OFFSET = 0;
+	private static final double CHANGE_OFFSET = 0;
+
+	private static final double TICKS_PER_REV = 2048;
+
+	private GenericEntry _entryAngles;
+
+	private double[] _angles = new double[2];
+
+	public Arm() {
+		_originMotor.configSelectedFeedbackSensor(SELECTED_SENSOR, 0, 10);
+		_changeMotor.configSelectedFeedbackSensor(SELECTED_SENSOR, 0, 10);
+
+		ShuffleboardTab tab = Shuffleboard.getTab("data");
+		_entryAngles = tab.add("Angles Of Rotation", new double[] {0, 0}).getEntry();
+	}
+
+	/**
+	 * Moves motors to set angles (pid?)
+	 * @param angles the angles to turn the motors (first = origin motor, second = change motor)
+	 */
+	public void moveAngles(double... angles) {
+		_originMotor.set(0.5);
+		while (true) {
```

Comment:
Very bad. This blocks the whole program.

Reply:
Nothing was changed here.
You SHOULD NOT block the program while the arm is moving.

Reply:
I don't see that, it looks like this for me
public void moveAngles(double speed, double... angles) {
		if (getOriginEncoder() <= angles[0]) {
			_originMotor.set(speed);
		}else{
			if (getChangeEncoder() >= angles[0]) {
				_changeMotor.set(speed);
			}else{
				_changeMotor.set(0);
			}
		}
	}